### PR TITLE
Fix not running test codes

### LINKED ISSUE #13085: DeepSpeed resume training test doesn't work
## 🐛 Bug

In [test_deepspeed_multigpu_stage_3_resume_training](https://github.com/PyTorchLightning/pytorch-lightning/blob/830da2c4a48570b114076024be2c24045593e592/tests/strategies/test_deepspeed_strategy.py#L797), test codes are located in `TestCallback`. But these codes aren't called.

### To Reproduce

Add a wrong condition to fail this test. But it succeeds.

```
$ git diff
diff --git a/tests/strategies/test_deepspeed_strategy.py b/tests/strategies/test_deepspeed_strategy.py
index 7fd13ae76..cf61752ac 100644
--- a/tests/strategies/test_deepspeed_strategy.py
+++ b/tests/strategies/test_deepspeed_strategy.py
@@ -820,6 +820,7 @@ def test_deepspeed_multigpu_stage_3_resume_training(tmpdir):
         def on_train_batch_start(
             self, trainer: Trainer, pl_module: LightningModule, batch: Any, batch_idx: int
         ) -> None:
+            assert False
             original_deepspeed_strategy = initial_trainer.strategy
             current_deepspeed_strategy = trainer.strategy
 
$ PL_RUN_STANDALONE_TESTS=1 python -m pytest -v tests/strategies/test_deepspeed_strategy.py -k test_deepspeed_multigpu_stage_3_resume_training
=================================================================== test session starts ====================================================================
platform linux -- Python 3.8.12, pytest-7.1.2, pluggy-1.0.0 -- /home/vscode/.venv/bin/python
cachedir: .pytest_cache
rootdir: /workspaces/pytorch/pytorch-lightning, configfile: setup.cfg
plugins: asyncio-0.18.3, forked-1.4.0, hydra-core-1.0.7, rerunfailures-10.2
asyncio: mode=legacy
collected 53 items / 31 deselected / 22 selected                                                                                                           

tests/strategies/test_deepspeed_strategy.py::test_deepspeed_multigpu_stage_3_resume_training PASSED                                                  [100%]

====================================================== 1 passed, 31 deselected, 15 warnings in 3.79s =======================================================
```
### Expected behavior
```
$ PL_RUN_STANDALONE_TESTS=1 python -m pytest -v tests/strategies/test_deepspeed_strategy.py -k test_deepspeed_multigpu_stage_3_resume_training

...

>       assert False
E       assert False

...

====================================================== 1 failed, 31 deselected, 15 warnings in 4.18s =======================================================
```
### Environment

- PyTorch Lightning Version : fab2ff35ad98ed021ee1d151bb67d3a4088c848b
- PyTorch Version : 1.11.0+cu113
- Python version : 3.8.12
- OS : Linux
- CUDA/cuDNN version: 11.6/8.2.4 
- GPU models and configuration: GeForce RTX 3060
- How you installed PyTorch : pip
- If compiling from source, the output of `torch.__config__.show()`:
- Any other relevant information:

### Additional context
The cause of this issue is specifying `fast_dev_run=True` and not specifying `max_epochs=2` to the trainer of resume training as its constructor arguments. So the current epoch of trainer starts with 1 and there are no iterations.

My solution to this issue is the following.

https://github.com/ar90n/pytorch-lightning/tree/bugfix/fix-not-runnnig-deepspeed-test

It includes the following modifications. The second modification is required for test codes running only once.

* Increase max_epochs in resume training
* Change hook from on_train_batch_start to on_train_epoch_start

If possible, please let me assign this issue.

cc @SeanNaren @awaelchli @rohitgr7 @akihironitta @borda

### PR DESCRIPTION
**Pull request by ar90n**
## What does this PR do?

Fixes #13085

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/strategies/test_deepspeed_strategy.py [hunk 1]
```diff
@@ -846,10 +844,10 @@ def on_train_batch_start(
     model = ModelParallelClassificationModel()
     trainer = Trainer(
         default_root_dir=tmpdir,
-        fast_dev_run=True,
         strategy=DeepSpeedStrategy(stage=3),
         accelerator="gpu",
         devices=1,
+        max_epochs=2,
```
Comment (awaelchli) [inline tests/strategies/test_deepspeed_strategy.py:850]: switch to max_steps=2?
```suggestion
        max_steps=2,
```

**Review by awaelchli — APPROVED** [review]

File: tests/strategies/test_deepspeed_strategy.py (hunk 1 above)
Comment (carmocca) [inline tests/strategies/test_deepspeed_strategy.py:850]: Since fast_dev_run was removed, we need to limit the # of batches to reduce the runtime

```suggestion
        max_epochs=2,
        limit_train_batches=1,
        limit_val_batches=0,
```

**Review by akihironitta — APPROVED** [review]
Once the comment above (https://github.com/PyTorchLightning/pytorch-lightning/pull/13089/files#r875999781) is resolved, LGTM!

**Comment by akihironitta** [comment]
@carmocca Could you check the milestone that this PR is added to when you get a chance? Its issue (https://github.com/PyTorchLightning/pytorch-lightning/issues/13085) is part of 1.7 but this PR is of 1.6.x. 👀

**Review by Borda — APPROVED** [review]

**Comment by ar90n** [comment]
CI has been busy all the time. Is there anything I should do? Why don't we restart it?

**Comment by akihironitta** [comment]
@ar90n There's no further action required, and we're aware of some issues with our CI. Thank you for checking.

**Review by carmocca — APPROVED** [review]

**Review by carmocca — APPROVED** [review]
